packages/core/src/utils/index.test.ts

```diff
@@ -0,0 +1,50 @@
+import {z} from "zod"
+import {formatZodErrors} from "./index"
+
+describe("formatZodErrors", () => {
+  const MySchema = z.object({
+    test: z.string(),
+  })
+
+  it("formats the zod error", () => {
+    const result = MySchema.safeParse({})
+    expect(formatZodErrors(result.error.format())).toEqual({test: "Required"})
```

Comment:
> Property 'error' does not exist on type '{ success: true; data: { test: string; }; } | { success: false; error: ZodError<{ test: string; }>; }'.

I'm not sure how to fix this.

Reply:
Adding this line before that will fix it.

```ts
if (result.success) throw new Error("Schema should not return success")
```

Typescript will use this to eliminate the `{ success: true; data: { test: string; }; }` from the next line